MyFridge/MyFridge/Refrigerator.cs

```diff
@@ -0,0 +1,235 @@
+            {
+                freeSpace += shelf.SpaceAvailable;
+            }
+            return freeSpace;
+        }
+
+        public void InsertingItem(Item item)
+        {
+            if (item == null) throw new Exception("invalid item");
+            if (GetFreeSpace()<item.Space) throw new Exception("There is not enough space");
+            foreach (var shelf in Shelves)
+            {
+                if (shelf.SpaceAvailable>=item.Space)
+                {
+                    shelf.InsertingItem(item);
+                    return;
+                }
+                throw new Exception("There is not enough space");
+
+            }
+  
+        }
+
+
+        public Item TookItem(int itemId)
+        {
+            Item empty;
+            foreach (var shelf in Shelves)
+            {
+                empty = shelf.TookItem(itemId);
+                if (empty != null) return empty;    
+
+            }
+             throw new Exception("No item if such ID number");
+           
+            
+        }
+
+        public void Cleaning()
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                shelf.Cleaning();
+            }
+        }
+
+        public List<Item> WantToEat(Kosher kosher, KindItem kind)
+        {
+            List<Item> items = new();
+            foreach (Shelf shelf in Shelves)
+            {
+                if (shelf.WantToEat(kosher,kind)!=null)
+                {
+                    items.AddRange(shelf.WantToEat(kosher, kind));
+                }
+            }
+            if (items.Count == 0) throw new Exception("Unfortunately there is currently no suitable food. ");
+            return items;
+        }
+        public List<Item> SortOfExpiryDate()
+        {
+            List<Item> items = new List<Item>();
+            foreach (var shelf in Shelves)
+            {
+                items.AddRange(shelf.Items);
+            }
+            items.Sort((p1, p2) => p1.ExpiryDate.CompareTo(p2.ExpiryDate));
+            
+            return items;
+
+        }
+        public List<Shelf> SortByAvailability()
+        {
+            List<Shelf> shelves= new();
+            shelves.AddRange(Shelves);
+            shelves.Sort((p1, p2) => p1.SpaceAvailable.CompareTo(p2.SpaceAvailable));
+            return shelves;
+
+        }
+        public List<Item> DairProducts(ref int freeSpace)
+        {
+        List <Item> items = new();
+            foreach (var shelf in Shelves)
+            {
+                foreach (var item in shelf.Items)
+                {
+                    if (item.Kosher == Kosher.Dairy && item.ExpiryDate.AddDays(-3) <= DateTime.Today)
+                    {
+                        freeSpace+=item.Space;
+                        items.Add(item);
+                    }
+                }
+            }
+            return items;
+        }
+        public void GoingShopping()
+        {
+            if (this.GetFreeSpace() >= 29)
```

Comment:
This is called "magic number". It's unclear what's it's purpose. 
Please use a const instead with a meaningful name.
This will make your code more robust as you'll also use the same const in line 156.